rmgpy/data/kinetics/family.py

```diff
@@ -1297,11 +1297,16 @@ def applyRecipe(self, reactantStructures, forward=True, unique=True):
                 productStructures.reverse()
 
         # If product structures are Molecule objects, update their atom types
+        # If product structures are Group objects and the reaction is in certain families
+        # (families with charged substances), the charge of structures will be updated
         for struct in productStructures:
             if isinstance(struct, Molecule):
                 struct.update()
             else:
                 struct.resetRingMembership()
+            if isinstance(struct, Group) and (label in ['1,2_insertion_co', 'r_addition_com', 'co_disproportionation',
```

Comment:
This could be lumped into the else above. You could even change it to `elif isinstance(struct, Group)` to eliminate the type assumption. E.g.:
```
if isinstance(struct, Molecule):
    struct.update()
elif isinstance(struct, Group):
    struct.resetRingMembership()
    if label in [...]:
        struct.update_charge()
else:
    raise TypeError('Expecting Molecule or Group object, not {0}'.format(struct.__class__.__name__))
```